src/main/java/taskalyn/TaskManager.java

```diff
@@ -0,0 +1,133 @@
+
+public class TaskManager {
+    private List<Task> tasks;
+    private Database database;
+    private Ui ui;
+
+    public TaskManager(Database database, Ui ui) {
+        this.database = database;
+        this.ui = ui;
+        this.tasks = new ArrayList<>(100);
+        loadDatabase();
+    }
+
+    public void addTask(Task task) {
+        tasks.add(task);
+        ui.printLines("Got it, I've added this task to your list!\n" +
+                "      " + task.toString() + "\n" + "    Wah bro... " + getTaskSize() + (getTaskSize() > 1 ? " tasks already!" : " task already!"));
+        updateDatabase();
+    }
+
+    public void deleteTask(int taskId) {
+        if (taskId <= tasks.size() + 1) {
+            Task task = tasks.get(taskId - 1);
+            tasks.remove(task);
+            ui.printLines("Awesome bro! One task gone :D\n" +
+                    "      " + task.toString() + "\n" + "    Wah bro... " + getTaskSize() + (getTaskSize() > 1 ? " tasks already!" : " task already!"));
+        }
+        updateDatabase();
+    }
+
+    public int getTaskSize() {
+        return tasks.size();
+    }
+
+    public void listTasks() {
+        String taskString = "Here are the tasks in your list:\n" + "    ";
+
+        if (!tasks.isEmpty()) {
+            System.out.println(tasks.size());
+            for (int i=0; i<tasks.size(); i++) {
+                System.out.println(i);
+                if (i != tasks.size() - 1) {
+                    taskString += i + 1 + "." + tasks.get(i).toString() + "\n    ";
+                }
+                else {
+                    taskString += i + 1 + "." + tasks.get(i).toString();
+                }
+            }
+        }
+        else {
+            taskString += "Nothing!";
+        }
+
+        ui.printLines(taskString);
+    }
+
+    public void completeTask(int taskId) {
+        if (taskId <= tasks.size() + 1) {
+            tasks.get(taskId - 1).setComplete();
+            ui.printLines("Nice, I've marked this task as complete:\n" +
+                    "       " + tasks.get(taskId - 1).toString());
+        }
+        updateDatabase();
+    }
+
+    public void incompleteTask(int taskId) {
+        if (taskId <= tasks.size() + 1) {
+            tasks.get(taskId - 1).setIncomplete();
+            ui.printLines("Ok, I've marked this task as incomplete:\n" +
+                    "       " + tasks.get(taskId - 1).toString());
+        }
+        updateDatabase();
+    }
+
+    private void loadDatabase() {
+        try {
+            List<String> txtLines = this.database.readFromDatabase();
+            for (String line : txtLines) {
+                Task task = findTask(line);
+                tasks.add(task);
+                System.out.println(tasks);
+            }
+        } catch (IOException e) {
+            System.out.println("Error reading database: " + e.getMessage());
+        }
+    }
+
+    private Task findTask(String line) {
+        String[] parts = line.split(" \\| ");
```

Comment:
I like that your list variables are plural

Reply:
Thanks!